app/controllers/application_controller.rb

```diff
@@ -60,4 +60,11 @@ def basecamp_auth_check
       @basecamp_agent = current_user.agents.where(type: 'Agents::BasecampAgent').first
     end
   end
+
+  def agent_params
+    params.permit!
```

Comment:
Why do you need the `params.permit!` on it's own line?

Does `params[:agent] && params[:agent].permit(:options, :memory, :name, :type, :schedule, :controller_ids, :control_target_ids ...) || {}` work?